test/MSBuildBinLogQuery.CommandLine/Program.cs

```diff
@@ -21,30 +22,79 @@ public static void Main(string[] args)
                 PrintErrorMessage($"File not found: {args[0]}");
             }
 
-            var events = new BinaryLogReader(args[0]).ReadEvents();
+            using var binaryLogReader = new BinaryLogReader(args[0]);
+            var events = binaryLogReader.ReadEvents();
 
             var graphBuilder = new GraphBuilder();
             graphBuilder.HandleEvents(events.ToArray());
 
-            PrintProjectNodes(graphBuilder.Projects.Values);
+            PrintProjectNodes(graphBuilder.Projects.Values.ToList());
         }
 
-        private static void PrintProjectNodes(IEnumerable<ProjectNode> projects)
+        private static void PrintProjectNodes(IEnumerable<Project> projects)
         {
             Console.WriteLine("projects:");
 
             foreach (var project in projects)
             {
                 Console.WriteLine($"  project #{project.Id}: {project.ProjectFile}");
+
+                // foreach (var target in project.Targets.Values)
+                // {
+                //     Console.WriteLine($"    target {target.Name}");
+                //     Console.WriteLine($"      directly before this: {string.Join(";", target.Node_BeforeThis.AdjacentNodes.Select(beforeThis => beforeThis.Name))}");
+                //     Console.WriteLine($"      directly after this: {string.Join(";", target.Node_AfterThis.AdjacentNodes.Select(afterThis => afterThis.Name))}");
+                // }
             }
 
             Console.WriteLine();
 
+            Console.WriteLine("project graph edges:");
+
             foreach (var project in projects)
             {
-                foreach (var beforeProject in project.ProjectsBeforeThis)
+                foreach (var beforeProject in project.Node_BeforeThis.AdjacentNodes)
+                {
+                    Console.WriteLine($"  #{project.Id} -> #{beforeProject.ProjectInfo.Id}");
```

Comment:
If you use this format, you can paste it into [Mermaid](https://mermaidjs.github.io/mermaid-live-editor) and get it nicely graphed for free :)
```suggestion
                    Console.WriteLine($"  P_{project.Id} --> P{beforeProject.ProjectInfo.Id}");
```

Reply:
Done.